airflow-core/src/airflow/ui/src/layouts/Details/DetailsLayout.tsx

```
@@ -68,6 +76,61 @@ export const DetailsLayout = ({ children, error, isLoading, tabs }: Props) => {
   const { onClose, onOpen, open } = useDisclosure();
   const [isRightPanelCollapsed, setIsRightPanelCollapsed] = useState(false);
 
+  const getDefaultSizes = useCallback(
+    (): PanelSizes => (dagView === "graph" ? DEFAULT_GRAPH_SIZES : DEFAULT_GRID_SIZES),
+    [dagView],
+  );
+
+  const getSavedSizes = useCallback((): PanelSizes => {
+    const storageKey = `panel-${dagId}-${dagView}`;
+
+    try {
+      const cached = sessionStorage.getItem(storageKey);
+
+      if (cached !== null && cached !== "") {
+        const parsed: unknown = JSON.parse(cached);
+
+        if (isPanelSizes(parsed)) {
+          return parsed;
+        }
+      }
+    } catch {
+      // Failed to parse saved panel sizes - fallback to default
+    }
```

I don't think we can have a failure reeading that. And if we do, that's not normal and we might want to break there instead of silently returning anything.